packages/expo-modules-core/android/src/main/java/expo/modules/interfaces/filesystem/ScopedDirectories.java

```diff
@@ -0,0 +1,43 @@
+package expo.modules.interfaces.filesystem;
+
+import java.io.File;
+import java.util.Objects;
+
+public final class ScopedDirectories {
+    private final File cacheDir;
+    private final File filesDir;
+
+    public ScopedDirectories(File cacheDir, File filesDir) {
+        this.cacheDir = cacheDir;
+        this.filesDir = filesDir;
+    }
+
+    public File getCacheDir() {
+        return cacheDir;
+    }
+
+    public File getFilesDir() {
+        return filesDir;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) return true;
+        if (obj == null || obj.getClass() != this.getClass()) return false;
+        var that = (ScopedDirectories) obj;
+        return Objects.equals(this.cacheDir, that.cacheDir) &&
+                Objects.equals(this.filesDir, that.filesDir);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(cacheDir, filesDir);
+    }
+
+    @Override
+    public String toString() {
+        return "ScopedDirectories[" +
+                "cacheDir=" + cacheDir + ", " +
+                "filesDir=" + filesDir + ']';
+    }
+}
```

Comment:
```suggestion
data class ScopedDirectories(val cacheDir: File, val filesDir: File)
```
By using Kotlin here, we get the same functionality in just one line of code.

Reply:
Sorry, still forgetting that I can mix and match kotlin and java and just followed the lang used in the template module I was using. This is actually codegen from Java record classes that is not yet in this Java version (i think), but kotlin data classes work great!